Lib/test/test_derived_stack_jy.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+from test import test_support
+
+class ExposedMethodStackRecursion(unittest.TestCase):
+
+    def test_java_str_repr_overrides(self):
+        """Py2kbuffer overrides __str__ and __repr__ directly in a PyObject subclass."""
+
+        # buffer interface translates to utf16-be on unicode input.
+        b=buffer(u'asdf')
+        self.assertEqual(str(b),b'\x00a\x00s\x00d\x00f')
+        # __str__ is exposed from Py2kBuffer
+        self.assertEqual(str(b), b.__str__())
+        # __repr__ is customized
+        self.assertRegexpMatches(repr(b),
+                                 r'<read-only buffer for 0x[\da-f]+, size -1, offset 0 at 0x[\da-f]+>')
+        # __repr__ is exposed
+        self.assertEqual(b.__repr__(),repr(b))
+
+    def test_derived_recursion(self):
+        """when PyObject exposes a method by it's actual name, the exposed facade delegates through
+        "  subclasses in PyObjectDerived and other ..Derived.  This fixes the above test, but opens
+        "  up a recursion issue.
+        """
+
+        class F(object):
```

Comment:
Nice test. Because `__str__` and `__repr__` cross-refer I wanted to check what happens if you only override one of them, so extending your test I wrote:
```
        class E(object):
            """This never passes through to super.__str__ or super.__repr__ in PyObjectDerived.java"""
            pass

        class F(object):
            def __repr__(self): return "myrepr"

        class G(object):
            def __str__(self): return "mystr"

        class H(object):
            def __repr__(self): return "myrepr"
            def __str__(self): return "mystr"

        for C in (E, F): # Both have original object.__str__
            C().__str__()
            str(C())

        for C in (E, G): # Both have original object.__repr__
            C().__repr__()
            repr(C())

        for C in (G, H): # Both override __str__
            self.assertEqual(C().__str__(), "mystr")
            self.assertEqual(str(C()), "mystr")

        for C in (F, H): # Both override __repr__
            self.assertEqual(C().__repr__(), "myrepr")
            self.assertEqual(repr(C()), "myrepr")
```